flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImplTest.java

```diff
@@ -120,15 +141,28 @@ public void testAbortIgnoresMissing() throws Exception {
         runWithSyncWorker(this::callAbort);
     }
 
+    @Test
+    public void testOldCheckpointIsAborted() {
+        ChannelStateWriterImpl writer = openWriter();
+        callStart(1, writer);
+        ChannelStateWriteResult result1 = writer.getWriteResult(1);
+        assertFalse(result1.isDone());
+
+        callStart(2, writer);
+        assertNull(writer.getWriteResult(1));
+        assertFalse(writer.getWriteResult(2).isDone());
+
+        assertWriteResultIsCompletedExceptionally(result1);
```

Comment:
Can you add assertion that the exception is `CheckpointException(CheckpointFailureReason#CHECKPOINT_DECLINED_SUBSUMED)`?